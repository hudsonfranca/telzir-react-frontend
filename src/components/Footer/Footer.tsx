import React from 'react';

import Container from './styles';
import Logo from '../../assets/Logo1.0.png';

const Footer: React.FC = () => {
  const currentYear = new Date().getUTCFullYear();

  return (
    <Container data-test="container">
      <img src={Logo} alt="Logo" />

      <small>
        &copy;
        {`Telzir ${currentYear}  All rights reserved.`}
      </small>
    </Container>
  );
};

export default Footer;
