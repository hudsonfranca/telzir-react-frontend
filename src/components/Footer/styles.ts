import styled from 'styled-components';

const Container = styled.div`
  grid-area: footer;
  width: 100%;
  height: 100%;
  background-color: var(--primary);
  font-size: 2rem;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  padding: 20px;
  position: relative;

  small {
    color: var(--white);
    font-size: 1.6rem;
  }

  &::before {
    content: '';
    width: 100%;
    height: 5px;
    position: absolute;
    top: 0;
    background-image: linear-gradient(
      to right,
      var(--primary),
      var(--quaternary),
      var(--quaternary),
      var(--primary)
    );
  }

  @media (max-width: 736px) {
    ul {
      flex-direction: column;
    }
  }
`;

export default Container;
