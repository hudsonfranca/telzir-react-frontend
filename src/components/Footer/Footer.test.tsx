import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import renderer from 'react-test-renderer';
import Footer from './Footer';
import 'jest-styled-components';

let wrapper: ShallowWrapper;

beforeEach(() => {
  wrapper = shallow(<Footer />);
});

describe('<Footer/> redering', () => {
  it('Should render without errors', () => {
    const footer = renderer.create(<Footer />).toJSON();

    expect(footer).toMatchSnapshot();
  });
});
