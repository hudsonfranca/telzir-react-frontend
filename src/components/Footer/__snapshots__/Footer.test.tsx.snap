// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> redering Should render without errors 1`] = `
.c0 {
  grid-area: footer;
  width: 100%;
  height: 100%;
  background-color: var(--primary);
  font-size: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
  position: relative;
}

.c0 small {
  color: var(--white);
  font-size: 1.6rem;
}

.c0::before {
  content: '';
  width: 100%;
  height: 5px;
  position: absolute;
  top: 0;
  background-image: linear-gradient( to right, var(--primary), var(--quaternary), var(--quaternary), var(--primary) );
}

@media (max-width:736px) {
  .c0 ul {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<div
  className="c0"
  data-test="container"
>
  <img
    alt="Logo"
    src="Logo1.0.png"
  />
  <small>
    Â©
    Telzir 2020  All rights reserved.
  </small>
</div>
`;
